pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go

```diff
@@ -154,10 +158,16 @@ func (r *ReconcilePerconaServerMongoDBRestore) reconcileRestore(cr *psmdbv1.Perc
 		return nil
 	}
 
-	bcpName := cr.Spec.BackupName
-	storageName := cr.Spec.StorageName
+	var (
+		backupName  = cr.Spec.BackupName
+		storageName = cr.Spec.StorageName
+	)
 
-	if bcpName == "" || storageName == "" {
+	if cr.Spec.PITR != nil && cluster.Spec.Backup.IsEnabledPITR() {
+		// getting the first and only storage if PITR is enabled
+		for storageName = range cluster.Spec.Backup.Storages {
+		}
```

Comment:
Can we use index instead?

Reply:
`cluster.Spec.Backup.Storages` is a map, so we can't use index :(

Reply:
Ok, then could you please do it more explicit
1. Comment says "getting the first and only storage if PITR is enabled", but you iterate over whole map
2. do not use `storageName` as a input var for key, but explicitly assign key to `storageName`

Reply:
BTW, if it's map we always will get different firs key

Reply:
> Ok, then could you please do it more explicit

changed in 337f664

> BTW, if it's map we always will get different firs key

Map len is always equal to 1 here, so order doesn't matter